components/camera/collectors.go

```diff
@@ -88,13 +87,18 @@ func newReadImageCollector(resource interface{}, params data.CollectorParams) (d
 		}
 	}
 
+	mimeStr := new(wrapperspb.StringValue)
+	if err := mimeType.UnmarshalTo(mimeStr); err != nil {
+		return nil, err
+	}
+
 	cFunc := data.CaptureFunc(func(ctx context.Context, _ map[string]*anypb.Any) (interface{}, error) {
 		_, span := trace.StartSpan(ctx, "camera::data::collector::CaptureFunc::ReadImage")
 		defer span.End()
 
 		ctx = context.WithValue(ctx, data.FromDMContextKey{}, true)
 
-		img, release, err := ReadImage(ctx, camera)
+		img, _, err := camera.Image(ctx, mimeStr.Value, nil)
```

Comment:
I'm pretty sure that it will break data capture for some users if the `extra` field is always nil. Previously [getExtra](https://github.com/viamrobotics/rdk/pull/4487/files#diff-4e03c7046f4cde2fb3cfabe60f410c7cc0380ec31028edec049f6a37cab65cd8L103) would parse out the extra fields from the context but we are no longer doing that.


I'll create a repro case to show how this breaks data capture.

Reply:
Please change this to:
```go
img, _, err := camera.Image(ctx, mimeStr.Value, data.FromDMExtraMap)
```

Reply:
> Please change this to:
> 
> ```go
> img, _, err := camera.Image(ctx, mimeStr.Value, data.FromDMExtraMap)
> ```

[Done](https://github.com/viamrobotics/rdk/pull/4487/commits/141b970dfc717b3447c44baef8e607907eb3169f)